src/main/java/com/cloudbees/jenkins/plugins/bitbucket/hooks/PushHookProcessor.java

```diff
@@ -44,10 +68,124 @@ public void process(String payload, BitbucketType instanceType) {
+                            if (!(source instanceof BitbucketSCMSource)) {
+                                return Collections.emptyMap();
+                            }
+                            BitbucketSCMSource src = (BitbucketSCMSource) source;
+                            if (!isBitbucketServerUrlMatch(src.getBitbucketServerUrl())) {
+                                return Collections.emptyMap();
+                            }
+                            if (!src.getRepoOwner().equalsIgnoreCase(getPayload().getRepository().getOwnerName())) {
+                                return Collections.emptyMap();
+                            }
+                            if (!src.getRepository().equalsIgnoreCase(getPayload().getRepository().getRepositoryName())) {
+                                return Collections.emptyMap();
+                            }
+                            BitbucketRepositoryType type = BitbucketRepositoryType.fromString(
+                                    getPayload().getRepository().getScm());
+                            if (type == null) {
+                                LOGGER.log(Level.INFO, "Received event for unknown repository type: {0}",
+                                        getPayload().getRepository().getScm());
+                                return Collections.emptyMap();
+                            }
+                            Map<SCMHead, SCMRevision> result = new HashMap<>();
+                            for (BitbucketPushEvent.Change change: getPayload().getChanges()) {
+                                if (change.isClosed()) {
+                                    result.put(new BranchSCMHead(change.getOld().getName(), type), null);
+                                } else {
+                                    BranchSCMHead head = new BranchSCMHead(change.getNew().getName(), type);
+                                    switch (type) {
+                                        case GIT:
+                                            result.put(head, new AbstractGitSCMSource.SCMRevisionImpl(head, change.getNew().getTarget().getHash()));
+                                            break;
+                                        case MERCURIAL:
+                                            result.put(head, new BitbucketSCMSource.MercurialRevision(head, change.getNew().getTarget().getHash()));
+                                            break;
+                                        default:
+                                            LOGGER.log(Level.INFO, "Received event for unknown repository type: {0}", type);
+                                            break;
+                                    }
+                                }
+                            }
+                            return result;
+                        }
+
+                        @Override
+                        public boolean isMatch(@NonNull SCM scm) {
+                            // TODO
```

Comment:
We are not implementing this at this point in time because we do not want to introduce the triggering of freestyle jobs based on the bitbucket hook data provided by the branch source registered hooks.

At a future point in time we will add the matching logic in here, but right now it would increase the scope of the change too much